int32 object_id
bool is_object_of_interest
Location[] visit_points